support-models/src/main/scala/com/gu/support/workers/ProductTypeRatePlans.scala

```diff
@@ -1,46 +1,48 @@
+  def productRatePlan[P: ProductTypeRatePlans](p: P, environment: TouchPointEnvironment, readerType: ReaderType): Option[ProductRatePlan[Product]] =
+    ProductTypeRatePlans[P].productRatePlan(p, environment, readerType)
 
   implicit class Ops[P: ProductTypeRatePlans](p: P) {
-    def productRatePlan(environment: TouchPointEnvironment, fixedTerm: Boolean): Option[ProductRatePlan[Product]] =
-      ProductTypeRatePlans[P].productRatePlan(p, environment, fixedTerm)
+    def productRatePlan(environment: TouchPointEnvironment, readerType: ReaderType): Option[ProductRatePlan[Product]] =
+      ProductTypeRatePlans[P].productRatePlan(p, environment, readerType)
   }
 
-  implicit val productTypeRatePlan: ProductTypeRatePlans[ProductType] = (product: ProductType, environment: TouchPointEnvironment, fixedTerm: Boolean) =>
+  implicit val productTypeRatePlan: ProductTypeRatePlans[ProductType] = (product: ProductType, environment: TouchPointEnvironment, readerType: ReaderType) =>
     product match {
-      case d: DigitalPack => d.productRatePlan(environment, fixedTerm)
-      case p: Paper => p.productRatePlan(environment, fixedTerm)
-      case w: GuardianWeekly => w.productRatePlan(environment, fixedTerm)
+      case d: DigitalPack => d.productRatePlan(environment, readerType)
+      case p: Paper => p.productRatePlan(environment, readerType)
+      case w: GuardianWeekly => w.productRatePlan(environment, readerType)
       case _ => None
     }
 
-  implicit val weeklyRatePlan: ProductTypeRatePlans[GuardianWeekly] = (product: GuardianWeekly, environment: TouchPointEnvironment, fixedTerm) => {
+  implicit val weeklyRatePlan: ProductTypeRatePlans[GuardianWeekly] = (product: GuardianWeekly, environment: TouchPointEnvironment, readerType) => {
     val postIntroductoryBillingPeriod = if (product.billingPeriod == SixWeekly) Quarterly else product.billingPeriod
     product.catalogType.ratePlans.getOrElse(environment, Nil).find(productRatePlan =>
       productRatePlan.fulfilmentOptions == product.fulfilmentOptions &&
         productRatePlan.billingPeriod == postIntroductoryBillingPeriod &&
-      productRatePlan.fixedTerm == fixedTerm
+      productRatePlan.readerType == readerType
     )
   }
 
   implicit val digitalRatePlan: ProductTypeRatePlans[DigitalPack] = (product: DigitalPack, environment: TouchPointEnvironment, fixedTerm) =>
     product.catalogType.ratePlans.getOrElse(environment, Nil).find(productRatePlan =>
-      (productRatePlan.billingPeriod == product.billingPeriod && productRatePlan.productOptions == product.productOptions) ||
-        (productRatePlan.productOptions == Corporate && product.productOptions == Corporate) // We don't care about the billing period for corporates
+      (productRatePlan.billingPeriod == product.billingPeriod && productRatePlan.readerType == product.readerType) ||
+        (productRatePlan.readerType == Corporate && product.readerType == Corporate) // We don't care about the billing period for corporates
```

Comment:
interestinlgly we will care about the billing period for gifts.

I would like to get rid of all these implicits and just use normal functions, a bit like we did with the productSubscriptionBuilders.  Something to keep in mind if we make significant changes to the code that uses them.

Reply:
Noted, thank you!